File: user-service/src/main/java/com/security/userservice/controller/user/UserController.java
package com.security.userservice.controller.user;

import com.security.userservice.domain.Account;
import com.security.userservice.domain.AccountDto;
import com.security.userservice.service.UserService;
import lombok.RequiredArgsConstructor;
import org.modelmapper.ModelMapper;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;

@Controller
@RequiredArgsConstructor
public class UserController {

    private final PasswordEncoder passwordEncoder;
    private final UserService userService;

    @GetMapping("/mypage")
    private String myPage() throws Exception {
        return "user/mypage";
    }

    @GetMapping("/users")
    public String createUser() {
        return "user/login/register";
    }

    @PostMapping("/users")
    public String createUser(AccountDto accountDto) {

        ModelMapper modelMapper = new ModelMapper();
        Account account = modelMapper.map(accountDto, Account.class);
        account.setPassword(passwordEncoder.encode(account.getPassword()));

        userService.createAccount(account);
        
        return "redirect:/";
    }
}

File: service/src/main/kotlin/com/example/service/controller/UserController.kt
package com.example.service.controller

import com.example.service.data.dto.UserDTO
import com.example.service.service.UserService
import org.springframework.web.bind.annotation.DeleteMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RequestMapping("/api/v1")
@RestController
class UserController(
        private val userService: UserService,
) {

    @PostMapping
    fun createUsers(userDTO: UserDTO) {
        userService.createUsers(userDTO)
    }

    @DeleteMapping
    fun deleteUsers(userId: String) {
        userService.deleteUsers(userId)
    }

}